src/Entity/Reflection/ModifierParser.php

```diff
@@ -190,29 +192,33 @@ private function processKeyword(string $value, ReflectionClass $reflectionClass)
 				$reflection = new ReflectionClass($className);
 			}
 
-			$enum = [];
-			$constants = $reflection->getConstants();
-			if (str_contains($const, '*')) {
-				$prefix = rtrim($const, '*');
-				$prefixLength = strlen($prefix);
-				$count = 0;
-				foreach ($constants as $name => $constantValue) {
-					if (substr($name, 0, $prefixLength) === $prefix) {
-						$enum[$constantValue] = $constantValue;
-						$count += 1;
-					}
-				}
-				if ($count === 0) {
-					throw new InvalidModifierDefinitionException("No constant matches $reflection->name::$const pattern.");
-				}
+			if ($reflection->isEnum() && is_subclass_of($className, BackedEnum::class)) {
+				return (new ReflectionEnum($className))->getCase($const)->getValue();
```

Comment:
Please utilize the early return, the original code will stay (not indented).
Also please add a test.

Reply:
done